libnetwork/network/interface.go

```diff
@@ -173,7 +174,7 @@ func getCniInterface(conf *config.Config) (types.ContainerNetwork, error) {
 		DefaultNetwork:     conf.Network.DefaultNetwork,
 		DefaultSubnet:      conf.Network.DefaultSubnet,
 		DefaultsubnetPools: conf.Network.DefaultSubnetPools,
-		IsMachine:          conf.Engine.MachineEnabled,
+		IsMachine:          machine.IsPodmanMachine(),
```

Comment:
Should also use IsGvProxyBased() although it shouldn't matter since this is only for pre 3.X backwards compat with the old extra podman-machine-cni plugin needed.

Reply:
Yeah I wasn't sure how the contract was used and the one usage seemed safe, so just made it generic. This is flipped to  GvProxy now.